### DISCUSSION TITLE
Builtin video capturing

### DISCUSSION THREAD

**Discussion by data-man** [contour-terminal/contour#278] [Ideas]
Before creating a detailed issue, I'd like to know different opinions. :smile:

**Comment by christianparpart** [comment]
I think that would be a very good fit for an extension. But for that we first need an extension architecture. And here my bar far most interesting choice is still to offer extensions via WebAssembly.

Such an extension could then hook into the [Terminal::ensureFreshRenderBuffer](https://github.com/contour-terminal/contour/blob/master/src/terminal/Terminal.cpp#L197) and read out the result.

p.s.: I really like using the Discussions for general discussions, so that tickets are actually action items / bug reports / feature requests.

**Reply by data-man** [reply]
> But for that we first need an extension architecture. 

Interestingly, plugin support has been added to Konsole (I'm using konsole-git). :)

**Reply by christianparpart** [reply]
Microsoft's Windows Terminal has a nice list of requirements that a extension ghost should support for terminals. I can link that here as soon as I am at a computer again.
